patterns/Factory-Design-Patten/README.md

```diff
@@ -0,0 +1,150 @@
+# Factory Design Pattern
+
+The Factory method is one of the most widely used design patterns. It functions as an interface for making objects that are modifiable after they are created. The difficulty creating objects without depending on the traditional constructor technique is addressed by this pattern. Rather, it creates the newly desired object based on the configuration or description that it receives (Cocca, 2022).
+
+
+In JavaScript, calling a constructor function with the new operator is a common way to create an object. Nevertheless, there are situations in which the client doesn't know or shouldn't be concerned with which of multiple possible objects to create. The value of the Factory Method resides in its ability to enable the client to assign object creation while retaining control over the kind of object that is instantiated (Poorte, 2023).
+
+## Diagram
+![javascript-factory-method](https://github.com/nazrinzuwair/pattern-library/assets/144160072/f5ab27dc-bd49-4c0b-9fed-92bce6f7df4d)
+
+The following entities are involved in this pattern when implementing using JavaScript:
+
+   - Creator: This corresponds to the 'factory' object in the provided visual representation, responsible for producing new products. It implements the 'factoryMethod', which yields freshly generated products.
+
+   - Concrete Product: These are the products being created, represented in the example code by the 'Employees' objects. Each product adheres to the same interface, comprising identical attributes and operations.
+
+**Abstract Product: This role is not commonly used in JavaScript. Traditionally, it would declare a product interface for various programming languages or scenarios. However, JavaScript does not specify interfaces explicitly. Instead, actual objects typically implement the relevant properties and methods directly, eliminating the need for an abstract product interface.**
+
+(Poorte, 2023)
+
+## Example demonstrating the Factory Method pattern in JavaScript:
+
+### Factory Function for Creating Foods
+
+The FoodFactory function serves as a factory for creating different types of foods, demonstrating the Factory Design Pattern. It contains a method createFood that takes a type as input and returns a corresponding food object. This approach abstracts the process of object creation, allowing the client code to create food objects without knowing the specific type being instantiated.
+
```

Comment:
As with the comment above - please make sure all inline reference to code in sentences or paragraphs are appropriately styled with single backticks.